File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anaDict = dict()
        
        for s in strs:
            t = ''.join(sorted(s))
            if t not in anaDict:
                anaDict[t] = []
            anaDict[t].append(s)
        
        return anaDict.values()
            
        

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List> strMap = new HashMap<>();

        for (String s: strs) {
            char[] c = s.toCharArray();
            Arrays.sort(c);
            String key = String.valueOf(c);
            if (!strMap.containsKey(key)) strMap.put(key, new ArrayList());
            strMap.get(key).add(s);
        }
        return new ArrayList(strMap.values());
    }
}